generators/client/templates/react/src/main/webapp/app/modules/home/home.tsx.ejs

```diff
@@ -18,110 +18,108 @@
 -%>
 import './home.<%= styleSheetExt %>';
 
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Link } from 'react-router-dom';
 import { Translate } from 'react-jhipster';
 import { connect } from 'react-redux';
 import { Row, Col, Alert } from 'reactstrap';
 
-import { IRootState } from 'app/shared/reducers';
 import { getSession } from 'app/shared/reducers/authentication';
 <%_ if (authenticationType === 'oauth2') { _%>
 import { getLoginUrl } from 'app/shared/util/url-utils';
 <%_ } _%>
 
 export interface IHomeProp extends StateProps, DispatchProps {}
 
-export class Home extends React.Component<IHomeProp> {
-  componentDidMount() {
-    this.props.getSession();
-  }
-
-  render() {
-    const { account } = this.props;
-    return (
-      <Row>
-        <Col md="9">
-          <h2><Translate contentKey="home.title">Welcome, Java Hipster!</Translate></h2>
-          <p className="lead"><Translate contentKey="home.subtitle">This is your homepage</Translate></p>
-          {
-            (account && account.login) ? (
-              <div>
-                <Alert color="success">
-                  <Translate contentKey="home.logged.message" interpolate={{ username: account.login }}>You are logged in as user {account.login}.</Translate>
-                </Alert>
-              </div>
-            ) : (
-              <div>
-                <Alert color="warning">
-                  <Translate contentKey="global.messages.info.authenticated.prefix">If you want to </Translate>
-                  <%_ if (authenticationType === 'oauth2') { _%>
-                  <a href={getLoginUrl()} className="alert-link">
-                    <Translate contentKey="global.messages.info.authenticated.link">sign in</Translate>
-                  </a>
-                  <%_ } else { _%>
-                  <Link to="/login" className="alert-link"><Translate contentKey="global.messages.info.authenticated.link"> sign in</Translate></Link>
-                  <%_ } _%>
-                  <Translate contentKey="global.messages.info.authenticated.suffix">, you can try the default accounts:
-                    <br />- Administrator (login=&quot;admin&quot; and password=&quot;admin&quot;)
-                    <br />- User (login=&quot;user&quot; and password=&quot;user&quot;).
-                  </Translate>
-                </Alert>
-
-                <%_ if (!skipUserManagement) { _%>
-                <Alert color="warning">
-                  <Translate contentKey="global.messages.info.register.noaccount">You do not have an account yet?</Translate>&nbsp;
-                  <Link to="/register" className="alert-link"><Translate contentKey="global.messages.info.register.link">Register a new account</Translate></Link>
-                </Alert>
+export const Home = (props: IHomeProp) => {
+  useEffect(() => {
+    props.getSession();
```

Comment:
I think the `getSession()` is just an extra api there since we retrieve the `account` data from the root component

Reply:
thanks @chrisdns it works, I remove the useEffect :)